tools/RAiDER/downloadGNSSDelays.py

```diff
@@ -271,6 +283,22 @@ def parse_years(timestr):
     return years
 
 
+def parse_cpus(cpustr):
+    '''
+    Takes string input and returns integer number of cpus for multiprocessing
+    '''
+    cpus = cpustr.lower()
+    # If user specifies 'all', pass maximum number of CPUs allowed by system. Otherwise, pass specified integer.
+    if cpus == 'all':
+        cpus = os.cpu_count()
+    elif cpus.isdigit():
+        cpus = int(cpus)
```

Comment:
Maybe some sanity checks like the number is positive. Don't know if you would want to put an upper bound on it.

Reply:
Not sure about the upper bound, the default of 8 should be fine and if a user specifies anything higher it is their own responsibility.

For a negative value, would you think it be better to raise an exception and exit, or just pass the absolute value? I hesitate with the latter because such an input error would be handled without transparency under the hood.

Reply:
Raise an exception. If you specify negative CPU's you should get an error

Reply:
Yes I agree, done.